internal/work/dbus_test.go

```diff
@@ -0,0 +1,61 @@
+package work
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestScrubName(t *testing.T) {
+	tests := []struct {
+		description string
+		input       string
+		want        string
+		wantError   error
+	}{
+		{
+			description: "no hyphens",
+			input:       "alpha",
+			want:        "alpha",
+		},
+		{
+			description: "single hyphen",
+			input:       "alpha-bravo",
+			want:        "alpha_bravo",
+			wantError:   ErrInvalidWorkerName,
```

Comment:
This should be changed to something like `InvalidDirectiveNameError{"alpha-bravo"}` and other cases too.

Reply:
This breaks the error test though, because the two errors are not the same. I had an implementation similar to what you're suggesting and the error tests didn't pass. I'm not *thrilled* with this solution either though. I did this only to allow for testing of the error cases. Maybe it's not worth defining a separate error type.

Reply:
Was `Is(target error)` function implemented correctly? When I change it to something like this:

```go
func (e WorkerError) Is(target error) bool {
	return e.Error() == target.Error()
}
```

then all tests passes with my suggested changes.

Reply:
Comparing the values of the `Error` function is considered bad practice. This is discussed in detail in [an issue](https://github.com/google/go-cmp/issues/289) on the go-cmp project.